processing/src/main/java/io/druid/segment/IndexMaker.java

```diff
@@ -542,7 +543,7 @@ private static void setupDimConversion(
 
       for (int i = 0; i < adapters.size(); i++) {
         Indexed<String> dimValues = adapters.get(i).getDimValueLookup(dimension);
-        if (dimValues != null) {
+        if (!IndexMerger.isNullColumn(dimValues)) {
```

Comment:
with the changes in this PR, DimDim treats null value as empty string and reports it as a dimValue so that queries can access that, so a null column can now have be the one having empty string with cardinality 1.